native/src/components/ConsentSection.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { ReactElement, useEffect, useState } from 'react'
+import { View, Switch } from 'react-native'
+import styled, { useTheme } from 'styled-components/native'
+
+import { ItemSeparator } from './Consent'
+import Text from './base/Text'
+
+const Container = styled(View)`
+  flex-direction: row;
+  padding: 16px;
+`
+
+const TextContainer = styled(View)`
+  flex: 1;
+`
+
+const Description = styled(Text)`
+  color: ${props => props.theme.colors.textSecondaryColor};
+`
+
+type ConsentSectionProps = {
+  title: string
+  description: string
+  initialSwitchValue: boolean
+  onPress: (type: string, value: boolean) => void
+}
+
+const ConsentSection = ({ title, description, initialSwitchValue, onPress }: ConsentSectionProps): ReactElement => {
+  const [allow, setAllow] = useState<boolean>(initialSwitchValue)
+
+  useEffect(() => {
+    setAllow(initialSwitchValue)
+  }, [initialSwitchValue])
```

Comment:
Do we actually need this? Or does this mean that this state is persisted twice (once here and once in the parent component/settings)? If that is the case, there should be no state in this component and the value should just be passed down (renamed to `allowed` or similar).

Reply:
ok i found the issue. i didn't check for externalSource to be not empty

Reply:
ok i had to add a check if externalPermissions where loaded before rendering the ConsentSections